test/compile_cache_test.rb

```diff
@@ -24,11 +24,17 @@ def test_coverage_running?
   end
 
   def test_no_write_permission_to_cache
-    path = Help.set_file('a.rb', 'a = 3', 100)
-    folder = File.dirname(Help.cache_path(@tmp_dir, path))
-    FileUtils.mkdir_p(folder)
-    FileUtils.chmod(0400, folder)
-    assert_raises(Bootsnap::CompileCache::PermissionError) { load(path) }
+    if RbConfig::CONFIG['host_os'] =~ /mswin|mingw|cygwin/
+      # Always pass this test on Windows, because it is not easily possible to make a directory read-only,
+      # i.e. FileUtils.chmod(0400, ...) has not the desired effect, neither does a call to system('attrib +R ...').
+      pass
+    else
```

Comment:
I'll ask around to see if there is an in-code way we can do this on Windows

Reply:
Well, there is the ACL api: You could remove the current user's access rights, e.g by using [SetSecurityInfo](https://docs.microsoft.com/en-us/windows/win32/api/aclapi/nf-aclapi-setsecurityinfo) But IMHO this is such a weird scenario on a Windows platform, it doesn't warrant that much effort. Still, if you find a good way, it would be very interesting for joeyates/imap-backup#78 😄

Reply:
After taking the time to read @daniel-rikowski's post and walking through the tests in this class, I think I understand the purpose for this test better, and this it doesn't apply on Windows. I've set the test to pass on Windows, as directories aren't 'read' like they are on Linux. You can restrict the ability to list files in a directory or restrict ACLs on the directory to prevent listing. Since the purpose of this is more to test we can still read files, I feel confident that skipping this test doesn't compromise testing reading files on Windows.